lib/Tools/LinearLayout.cpp

```diff
@@ -955,6 +955,17 @@ LinearLayout LinearLayout::invertAndCompose(const LinearLayout &outer) const {
   return ret;
 }
 
+LinearLayout LinearLayout::invert() const {
+  // A^-1(x) = A^-1(I(x)), thus A.invert() = I.invertAndCompose(A)
+  assert(isSurjective() && getTotalInDimSize() == getTotalOutDimSize() &&
```

Comment:
I think this is interesting enough that can be factored out into `LinearLayout::isInvertible`

Reply:
Done